src/trace-writer.js

```diff
@@ -186,10 +202,15 @@ TraceWriter.prototype.writeSpan = function(spanData) {
 TraceWriter.prototype.queueTrace_ = function(trace) {
   var that = this;
 
+  if (!that.isActive) {
+    return;
```

Comment:
One option would be for the `bind` and `bindEmitter` operations to just no-op if the agent is disabled. This wouldn't be a nice as short-circuiting out of the plugin instrumentation but may avoid the majority of the performance overhead. It would be worth thinking about the design of this disabling mechanism without plugin unpatching in more detail to make sure we're getting it right.